lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -372,81 +358,85 @@ static Optional<LineColumnRange> parseExpectedFixItRange(
+    unsigned FirstVal = 0;
+    if (Str.consumeInteger(10, FirstVal)) {
+      if (LineOffsetKind == OffsetKind::None) {
         diagnoseError(Str.data(),
-                      "expected line offset after leading '+' or '-' in fix-it "
-                      "verification");
+                      "expected line or column number in fix-it verification");
       } else {
         diagnoseError(Str.data(),
-                      "expected line or column number in fix-it verification");
+                      "expected line offset after leading '+' or '-' in fix-it "
+                      "verification");
       }
       return None;
     }
 
-    unsigned secondNumber = LineColumnRange::NoValue;
-    if (!Str.empty() && Str.front() == ':') {
-      Str = Str.drop_front();
-
-      if (Str.consumeInteger(10, secondNumber)) {
-        diagnoseError(
-            Str.data(),
-            "expected column number after ':' in fix-it verification");
-        return None;
+    // If the first value is not followed by a colon, it is either a column or a
+    // line offset that is missing a column.
+    if (Str.empty() || Str.front() != ':') {
+      if (LineOffsetKind == OffsetKind::None) {
+        return ParsedLineAndColumn{None, FirstVal};
       }
-    } else if (lineOffsetKind > LineOffsetKind::None) {
+
       diagnoseError(Str.data(),
                     "expected colon-separated column number after line offset "
                     "in fix-it verification");
       return None;
     }
 
-    if (secondNumber == LineColumnRange::NoValue) {
-      // If only one value is specified, it's a column number;
-      return std::make_pair(LineColumnRange::NoValue, firstNumber);
+    unsigned Column = 0;
+    Str = Str.drop_front();
+    if (Str.consumeInteger(10, Column)) {
+      diagnoseError(Str.data(),
+                    "expected column number after ':' in fix-it verification");
+      return None;
     }
 
-    unsigned lineNo = DiagnosticLineNo;
-    switch (lineOffsetKind) {
-    case LineOffsetKind::None:
-      lineNo = firstNumber;
+    // Apply the offset relative to the line of the expected diagnostic.
+    switch (LineOffsetKind) {
+    case OffsetKind::None:
       break;
-    case LineOffsetKind::Plus:
-      lineNo += firstNumber;
+    case OffsetKind::Plus:
+      FirstVal += DiagnosticLineNo;
       break;
-    case LineOffsetKind::Minus:
-      lineNo -= firstNumber;
+    case OffsetKind::Minus:
+      FirstVal = DiagnosticLineNo - FirstVal;
       break;
     }
 
-    return std::make_pair(lineNo, secondNumber);
+    return ParsedLineAndColumn{FirstVal, Column};
   };
 
   LineColumnRange Range;
 
-  if (const auto lineAndCol = parseLineAndColumn()) {
-    std::tie(Range.StartLine, Range.StartCol) = lineAndCol.value();
+  if (const auto LineAndCol = parseLineAndColumn()) {
+    // The start line defaults to the line of the expected diagnostic.
+    Range.StartLine = LineAndCol->Line.getValueOr(DiagnosticLineNo);
```

Comment:
```suggestion
    Range.StartLine = LineAndCol->Line.value_or(DiagnosticLineNo);
```
`getValueOr` is deprecated.

Reply:
Thanks!